pkg/controller/subscription/subscription_controller.go

```diff
@@ -253,24 +244,27 @@ func (r *ReconcileSubscription) HandleUpgrades(ctx context.Context, rhmiSubscrip
 
 	isServiceAffecting := rhmiConfigs.IsUpgradeServiceAffecting(latestRHMICSV)
 
-	// checks if there is changes in the rhmiconfig cr and recalculate the status
-	if reschedule, ok := config.Annotations[v1alpha1.RecalculateScheduleAnnotation]; ok && reschedule == "true" {
-		err = rhmiConfigs.UpdateStatus(ctx, r.client, config, latestRHMIInstallPlan, rhmiSubscription.Status.CurrentCSV)
-		if err != nil {
+	if isServiceAffecting {
+
+		config.Status.UpgradeAvailable = &integreatlyv1alpha1.UpgradeAvailable{
+			TargetVersion: rhmiSubscription.Status.CurrentCSV,
+			AvailableAt:   latestRHMIInstallPlan.CreationTimestamp}
+
+		if err := r.client.Status().Update(context.TODO(), config); err != nil {
```

Comment:
```suggestion
	if isServiceAffecting {
                newUpgradeAvailable := &integreatlyv1alpha1.UpgradeAvailable{
			TargetVersion: rhmiSubscription.Status.CurrentCSV,
			AvailableAt:   latestRHMIInstallPlan.CreationTimestamp
	        }
	        
	        if !reflect.DeepEquals(newUpgradeAvailable, config.Status.UpgradeAvailable {
	                config.Status.UpgradeAvailable = newUpgradeAvailable
		        if err := r.client.Status().Update(context.TODO(), config); err != nil {
		                return reconcile.Result(), err
		        }
		        return reconcile.Result{RequeueAfter: time.Second}
		}
```

I think there might be a race condition as the RHMIConfig controller (running concurrently) might have not updated the status when the verification for whether the upgrade must be approved or not is performed. I'd like to test this to ensure this, but if there is a race condition this suggestion might solve it. Update the UpgradeAvailable field and if it has changed, requeue the reconciliation after a second, giving the RHMIConfig controller enough time to reconcile the status.

Reply:
Thanks Sergio, didn't realize until now that this could be happening! Thanks a lot for clarification!